Wbooks/src/redux/book/actions.js

```diff
@@ -15,12 +21,37 @@ const BookActions = {
       book
     }
   }),
-  getBookList: () => ({
-    type: actionTypes.GET_BOOK_LIST
-  }),
-  getRentedBooks: () => ({
-    type: actionTypes.GET_RENTED_BOOKS
-  }),
+  getBookList: () => async dispatch => {
+    dispatch({ type: actionTypes.GET_BOOK_LIST });
+    const response = await BookService.getBookList();
+    console.log(response);
+    if (response.ok) {
+      dispatch({
+        type: actionTypes.GET_BOOK_LIST_SUCCESS,
+        payload: response.data.page
+      });
+    } else {
+      dispatch({
+        type: actionTypes.GET_BOOK_LIST_ERROR
+      });
```

Comment:
Remember that you can get the `response.error` or wherever the error code or message could be on the response and send it to the reducer and then get the error from any connected component to response to that particular error. Again, it's just a reminder